Managers/ConsoleMan.cpp

```diff
@@ -1,541 +1,423 @@
-            m_LastLogMove = 1;
-
-            // Avoid falling off the end
-            if (m_InputLogPosition == m_InputLog.end())
-            {
-                m_InputLogPosition--;
-                m_LastLogMove = 0;
-            }
-        }
-        // Down pressed! Load next logged input string into the input line
-        else if (g_UInputMan.KeyPressed(KEY_DOWN))
-        {
-            // See if we should decrement doubly because the last move was in the opposite direction
-            if (m_LastLogMove > 0 && m_InputLogPosition != m_InputLog.begin())
-                m_InputLogPosition--;
-
-            // If already at the beginning, then only put in empty string
-            if (m_InputLogPosition == m_InputLog.begin())
-            {
-                m_pInputTextBox->SetText("");
-                m_LastLogMove = 0;
-            }
-            else
-            {
-                m_InputLogPosition--;
-                // Revive logged input and put cursor at the end of it
-                m_pInputTextBox->SetText(*m_InputLogPosition);
-                m_pInputTextBox->SetCursorPos(m_pInputTextBox->GetText().length());
-                m_LastLogMove = -1;
-            }
-        }
-    }
-
-// TODO: This sometime.. bleh
-    // Cut off the text in the text label at a reasonable height so it doesn't get really slow to draw
-//    if (m_pConsoleText->GetTextHeight() > g_FrameMan.)
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          Draw
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Draws this ConsoleMan's current graphical representation to a
-//                  BITMAP of choice.
-
-void ConsoleMan::Draw(BITMAP *pTargetBitmap)
-{
-    if (IsVisible() || m_ForceVisible)
-    {
-        AllegroScreen drawScreen(pTargetBitmap);
-        m_pGUIController->Draw(&drawScreen);
-    }
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          ClearLog
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Clears all previous input.
-
-void ConsoleMan::ClearLog()
-{
-    m_InputLog.clear();
-    m_InputLogPosition = m_InputLog.begin();
-    m_pConsoleText->SetText("");
-}
-
-
-} // namespace RTE
+
+		m_InputTextBox->SetText("");
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::ConsoleOpenClose() {
+		float toGo = 0;
+		switch (m_ConsoleState) {
+			case ENABLING:
+				m_ParentBox->SetEnabled(true);
+				m_ParentBox->SetVisible(true);
+
+				toGo = -std::floorf(static_cast<float>(m_ParentBox->GetYPos()) * 0.5F);
+				m_ParentBox->SetPositionAbs(0, m_ParentBox->GetYPos() + toGo);
+
+				if (m_ParentBox->GetYPos() >= 0) { m_ConsoleState = ENABLED; }
+				break;
+			case ENABLED:
```

Comment:
I think maybe something's not right here? The only code I see that calls ConsoleOpenClose is in Update and it looks like
`if (m_ConsoleState != ENABLED && m_ConsoleState != DISABLED) { ConsoleOpenClose(); }`

Given you're breaking after each case (which is obviously good), how could state ever be enabled or disabled in this method?

Reply:
Hmm you're right, the ENABLED and DISABLED cases never run, my bad.
But, if they're never running and the console appears to be functioning as intended, this means that I actually don't need them at all.
So i'm just gonna go ahead and yeet them.

Reply:
As long as you're sure you didn't screw something up (i.e. it shouldn't be in that if check) then sure, yeet away.